packages/bitcore-node/migrations/fixUnspentInputs.js

```diff
@@ -52,51 +56,106 @@ class Migration {
-          );
-          console.log('modified count: ', task.modifiedCount);
-        }
-        output[`${chain}-${network}`] = recordsToUpdate;
-      }
-    }
-    const date = new Date().getTime();
-    const filename = `output-${date}.log`;
-    console.log(`writing output to ${filename}`);
-    try {
-      await fsPromises.writeFile(filename, JSON.stringify(output));
-    } catch (e) {
-      // write to stdout
-      console.log('failed to write output to file. Writing to stdout instead.');
-      console.log(output);
+    const { chain, network, dryrun } = args;
+    console.log(`Checking records for ${chain}:${network}`);
+    // Get all pending coins from valid transactions (mintHeight should be valid block height)
+    const stream = this.coinModel.collection
+      .find(
+        { chain, network, mintHeight: { $gt: -1 }, spentHeight: -1 } // -1 is pending status
+      )
+      .addCursorFlag('noCursorTimeout', true);
+
+    // Initialize RPC connection
+    if (!Config.default.chains[chain]) {
+      console.error(`There is no config for chain '${chain}'`);
+      return;
     }
-    if (args.dryrun) {
-      console.log('run the script with "--dryrun false" to execute this operation on the returned results.');
+    if (!Config.default.chains[chain][network]) {
+      console.error(`There is no config for chain '${chain}' with network '${network}'`);
+      return;
     }
+    const rpcConfig = Config.default.chains[chain][network].rpc;
+    const rpc = new CryptoRpc(
+      {
+        rpcPort: rpcConfig.port,
+        host: rpcConfig.host,
+        protocol: rpcConfig.protocol,
+        rpcUser: rpcConfig.username,
+        rpcPass: rpcConfig.password,
+        chain
+      },
+      {}
+    ).get(chain);
+
+    const handleStream = async data => {
+      let isUnspent = false;
+      // If spent (or in mempool) then this returns null with an error otherwise returns data on unspent output
+      try {
+        const coinData = await rpc.getTxOutputInfo({
+          txid: data.mintTxid,
+          vout: data.mintIndex
+        });
+        isUnspent = !!coinData;
+      } catch (e) {
+        // Coin must be spent or actually pending in mempool - do nothing
+      } finally {
+        if (isUnspent) {
+          // Log record
+          if (output[`${chain}-${network}`]) {
+            output[`${chain}-${network}`].push(data);
+          } else {
+            output[`${chain}-${network}`] = [data];
+          }
+
+          if (!dryrun) {
+            // Update record to be unspent (-2)
+            this.coinModel.collection.updateOne({ _id: data._id }, { $set: { spentHeight: -2 } }); // -2 is unspent status
+          }
+        }
+        const nextData = await stream.next();
+        if (nextData) {
+          handleStream(nextData);
+        } else {
+          // End of data -
+          endProcess();
+        }
+      }
+    };
+    // Handle incoming stream data
+    handleStream(await stream.next());
```

Comment:
need to handle if `await stream.next()` returns null here.

Reply:
Nice catch.